package/src/utilities/add-vite-plugin.ts

```diff
@@ -15,17 +27,47 @@ import type { PluginOption } from "vite";
  * ```ts
  * addVitePlugin({
  * 		plugin,
+ * 		config,
+ * 		logger,
  * 		updateConfig
  * })
  * ```
  */
 export const addVitePlugin = ({
+	warnDuplicated = true,
 	plugin,
+	config,
+	logger,
 	updateConfig,
-}: {
-	plugin: PluginOption;
-	updateConfig: HookParameters<"astro:config:setup">["updateConfig"];
-}) => {
+}:
+	| {
+			warnDuplicated: false;
```

Comment:
I moved this out into some interfaces, but the type errors are not as descriptive as they were when I had it setup like this. Should I revert it to this so that users have better errors when passing the wrong types?